### PR TITLE
Signup flow api hookup

### PR DESCRIPTION
**Pull request by LandGod**
Features included in this PR:
- Info for current user on signup page now pulled from sessionStorage
- Hitting submit on create household form now does the following:
-- Creates household document
-- Creates (or updates) documents for each user specified, including household id
-- Adds household Id to session storage
-- Adds all user ids to household members array
-- Adds user's objectId to session storage
-- Redirects to dashboard on success

Notes:
The way new users are currently added to the database may cause problems or break unexpectedly if the user already existed. To that end the associated functions should be retooled eventually to account for this. 

Also, in general, this workflow is kinda fragile. A nice to have would be some sort of recovery code that nukes the changes to the database if we fail out somewhere, since we could end up with some weird stuff if that happens. This will be a LOT of work to do right though, so probably something to add to the very bottom of our nice-to-have list.

Signing in does not currently redirect to the createhousehold page for new users. (I believe that functionality is on Jenn's branch?) To test signup flow, simply login, wait a moment for session storage to populate, and then manually navigate to "/signup". 

Because the file I was using for utility functions between controllers is not accessible from within the client folder, I added a folder to components called utility functions to server a similar purpose. It currently contains only one function: which pulls user data from session storage. 

A good chunk of apparent changes to member.js are just formatting. Not really sure what happened there. I did also add some methods to the file as well though.

### CODE REVIEW

File: client/src/components/utilityFunctions/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  // Get user info object from client storage
+  // The oauthKey can be accessed from the returned object using [oauthKey] or [id] to allow for
+  // personal preference and avoid confusion (since it is called different things in different places)
+  getLocalUserInfo: function() {
+    return {
+      firstName: sessionStorage.getItem("firstName"),
+      lastName: sessionStorage.getItem("lastName"),
+      email: sessionStorage.getItem('email'),
+      oauthKey: sessionStorage.getItem("id"),
+      id: sessionStorage.getItem("id")
```
Comment (jennhenn18) [inline client/src/components/utilityFunctions/index.js]: this should be userID

File: routes/api/household.js [hunk 2]
```diff
@@ -59,4 +59,38 @@ router
...
+
+  })
+
+  router.route('/add-members')
+  .put(function(req, res){
+
+    // Validate
+    if (!req.body) {res.status(400).send('Request had no body!'); return};
+    if (!req.body.householdId) {res.status(400).send('No household id!'); return};
+    if (!req.body.idsArray || req.body.idsArray.lenght < 1) {res.status(400).send('Array of member ids was empty or undefined!'); return};
```
Comment (jennhenn18) [inline routes/api/household.js]: length is spelled wrong here

File: routes/api/member.js [hunk 3]
```diff
@@ -4,89 +4,95 @@ const mongoose = require("mongoose"); // For working with mongoose Date and Obje
...
 
-    });
+router.route("/login").put(function(req, res) {
+  memberController
+    .findByEmail(req.body.email)
+    .then(function(result) {
+      // if no email exists, return 204 status message to client. This will tell the client to redirect to the household page
+      if (result.length === 0) {
+        res.status(204).end();
+        return;
```
Comment (jennhenn18) [inline routes/api/member.js]: do we want to add an error capture here?

File: client/src/components/utilityFunctions/index.js (hunk 1 above)
Comment (LandGod) [reply client/src/components/utilityFunctions/index.js]: Done!

File: routes/api/household.js (hunk 2 above)
Comment (LandGod) [reply routes/api/household.js]: Fixed! Good catch!

**Pull request merged**